src/wallet/scriptpubkeyman.cpp

```diff
@@ -1091,6 +1092,7 @@ void LegacyScriptPubKeyMan::KeepDestination(int64_t nIndex, const OutputType& ty
     // Remove from key pool
     WalletBatch batch(m_storage.GetDatabase());
     batch.ErasePool(nIndex);
+    LearnRelatedScripts(pubkey, type);
```

Comment:
In commit "Key pool: Move LearnRelated and GetDestination calls" (094f0545d2d1309ad85eb66f107dec52746490df)

Commit description "There is a minor change in behavior" seems out of date. Would just say "This commit does not change behavior".

The previous description was referring to LearnRelatedScripts call happening moving under the GetReservedDestination `nIndex == -1` check in the old version of this PR before #17237.

Reply:
Fixed.

Reply:
Note: This change was done in the wrong commit "Change ReturnDestination interface" (82dbc9344c33217e01c16b27d1418f042d0d0d77) instead the suggested "Move LearnRelated and GetDestination" (393315354da454eb0f9322b92ad736bd23606f97), so `dest` return value is temporarily left unset by the earlier commit and then fixed again in a later unrelated commit. Would be nice to fix but not important.